src/d3a/models/area/redis_external_connection.py

```diff
@@ -11,8 +11,32 @@ def __init__(self, area):
         self.pubsub = self.redis_db.pubsub()
         self.sub_to_area_event()
         self.areas_to_register = []
+        self.areas_to_unregister = []
+
+    def market_cycle_event(self):
+        self.unregister_pending_areas()
```

Comment:
I guess that you dont need this here as it is already part of `register_new_areas`. right?

Reply:
You are right, I forgot to remove it from the register_new_areas method, doing it. Thanks for pointing it out.